controllers/storagecluster/cephnfs.go

```diff
@@ -0,0 +1,340 @@
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("uninstall: unable to retrieve CephBlockPool %v: %v", cephBlockPool.Name, err)
+	}
+
+	if cephBlockPool.GetDeletionTimestamp().IsZero() {
+		r.Log.Info("Uninstall: Deleting CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+		err = r.Client.Delete(context.TODO(), foundCephBlockPool)
+		if err != nil {
+			r.Log.Error(err, "Uninstall: Failed to delete CephBlockPool.", "CephBlockPool", klog.KRef(foundCephBlockPool.Namespace, foundCephBlockPool.Name))
+			return reconcile.Result{}, fmt.Errorf("uninstall: Failed to delete CephBlockPool %v: %v", foundCephBlockPool.Name, err)
+		}
+	}
+
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: cephBlockPool.Name, Namespace: sc.Namespace}, foundCephBlockPool)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Uninstall: CephBlockPool is deleted.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+			return reconcile.Result{}, nil
+		}
+	}
+	r.Log.Error(err, "Uninstall: Waiting for CephBlockPool to be deleted.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
+	return reconcile.Result{}, fmt.Errorf("uninstall: Waiting for CephBlockPool %v to be deleted", cephBlockPool.Name)
+}
+
+type ocsCephNFSService struct{}
+
+// newNFSService returns the Service instance that should be created on first run.
+func (r *StorageClusterReconciler) newNFSService(initData *ocsv1.StorageCluster) (*v1.Service, error) {
+	obj := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForNFSService(initData),
+			Namespace: initData.Namespace,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name: "nfs",
+					Port: 2049,
+				},
+			},
+			Selector: map[string]string{
+				"app":      "rook-ceph-nfs",
+				"ceph_nfs": generateNameForCephNFS(initData),
+			},
+			SessionAffinity: "ClientIP",
+		},
+	}
+
+	err := controllerutil.SetControllerReference(initData, obj, r.Scheme)
+	if err != nil {
+		r.Log.Error(err, "Unable to set Controller Reference for NFS service.", " NFSService ", klog.KRef(obj.Namespace, obj.Name))
+		return nil, err
+	}
+
+	return obj, nil
+}
+
+// ensureCreated ensures that cephNFS related service resource exist in the desired state.
+func (obj *ocsCephNFSService) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if instance.Spec.NFS == nil || !instance.Spec.NFS.Enable {
+		return reconcile.Result{}, nil
+	}
+
+	nfsService, err := r.newNFSService(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	existingNFSService := v1.Service{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: nfsService.Name, Namespace: nfsService.Namespace}, &existingNFSService)
+	switch {
+	case err == nil:
```

Comment:
default case is required for others errors

Reply:
> default case is required for others errors

added